d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadLoadBalancer.java

```diff
@@ -116,6 +116,10 @@ public void start(Callback<None> callback)
       _rateLimitedLogger.debug("newLb executor rejected new task for start. "
           + "It is shut down or its queue size has reached max limit");
     }
+    catch (Exception e)
+    {
+      _rateLimitedLogger.debug("Failed to execute newLb task for start. ", e);
+    }
```

Comment:
We don't want to write restli code according to particular class passed in from container. It's better to guard it in general.

Reply:
`RejectedExecutionHandler` and `RejectedExecutionException` are only `ThreadPoolExecutor` constructs, so this code is already assuming that a `ThreadPoolExecutor` is being used. If the reference to the executor was to `ThreadPoolExecutor` rather than `ExecutorService`, then it would've made sense to handle this exception. Otherwise, not.

Reply:
It depends on the actual handler used to handle the rejection, doesn't it? According to [ThreadPoolExecutor's doc](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadPoolExecutor.html) : "Four predefined handler policies are provided:
In the default [ThreadPoolExecutor.AbortPolicy](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadPoolExecutor.AbortPolicy.html), the handler throws a runtime [RejectedExecutionException](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/RejectedExecutionException.html) upon rejection."

Reply:
> We don't want to write restli code according to a particular class passed in from container. It's better to guard it in general.

The point is that this statement is likely incorrect, since `RejectedExecutionException` only needs to be handled when the reference is to `ThreadPoolExecutor`. In this case, in Restli, the object reference is to `ExecutorService` instead, which doesn't throw this exception.

Reply:
oh, I see. Yes, you're right. That makes sense.

Reply:
@brycezhongqing did discuss this with me earlier and it's not necessary to make that change though (it's up to him). Just that we can avoid adding more exception handling for now perhaps.

Reply:
Yes. We could not catch the `ThreadPoolExecutor` task exception here.  I will raise a new PR focus on removing the extra exception and add more exception inside the task of `ThreadPoolExecutor`